.github/workflows/generate-ts-declaration.yml

```
@@ -0,0 +1,69 @@
+name: generate-typescript-declaration
+run-name: generate-typescript-declaration
+on:
+  push:
+    paths:
+      - '**.schema.json'
+
+jobs:
+  generate-typescript-declaration:
+    runs-on: ubuntu-latest
+    name: generate-typescript-declaration
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v3
+
+      # Generate application schemas to TypeScript declarations
+      - name: Generate application schemas to TypeScript declarations
+        id: tag
+        uses: ./.github/actions/generate-typescript-declaration
+        with:
+          schema-directory: ./schemas/applications/schema
+          schema: application.schema.json
+          ts-declaration: application.d.ts
+
+      # Checkout to frontend repository and create branch ci/update-schemas
+      - name: Checkout frontend repository
+        uses: actions/checkout@v3
+        with:
+          repository: aica-technology/frontend
+          ref: develop
+          path: frontend
+          token: ${{ secrets.FRONTEND_REPOSITORY_WRITE }}
+      - name: Create a new branch in frontend repository
+        run: |
+          cd frontend
+          ls
+          git config --global user.name github-actions[bot]
+          git config --global user.email github-actions[bot]@users.noreply.github.com
+
+          git checkout -b ci/update-schemas
+          git push -u -f origin ci/update-schemas
```

Since you already checkout the other repo and push a branch here, it seems it would be more efficient to just add the new generated files in this step. That way you can skip the two `write-file` steps afterwards.

You would first have to do both `generate-typescript-declaration` actions, move the results manually, then just add and commit.

```suggestion
          mv schemas/applications/schema/application.d.ts frontend/src/utils/schemas/
          mv schemas/component-descriptions/schema/dynamic-component-description.d.ts frontend/src/utils/schemas/

          cd frontend
          git config --global user.name github-actions[bot]
          git config --global user.email github-actions[bot]@users.noreply.github.com

          git checkout -b ci/update-schemas
          git add src/utils/schemas
          git commit -m "ci: update typescript declaration files from new schemas"
          git push -u -f origin ci/update-schemas
```

It's nice to re-use the write-file action where possible but for this particular case it's not well suited because we require a new feature branch on the remote repo first.